torchvision/models/detection/roi_heads.py

```diff
@@ -525,6 +527,15 @@ def forward(self, features, proposals, image_shapes, targets=None):
             image_shapes (List[Tuple[H, W]])
             targets (List[Dict])
         """
+        if targets is not None:
+            for t in targets:
+                assert t["boxes"].dtype == torch.float32, 'target boxes must of float type'
```

Comment:
Those checks might not work if we want to perform fp16 training in the future.
Maybe might be better to use something like `t["boxes"].dtype.is_floating_point`?

Reply:
Yes, you are right! Will change it.